### DISCUSSION TITLE
Does the certstore "--keep" flag ensure the downloaded certificate is available in the kernel environment?

### DISCUSSION THREAD

**Discussion by gregorybruno** [ipxe/ipxe#466] [Q&A]
Here's the use case:

- The ipxe script uses "certstore" to download a certificate in order to use https to download other files (e.g., the linux kernel, initrd, etc.).
- I would also like to use that same certificate in the kernel environment (that is, after the ipxe script executes "boot").

It seems that the "--keep" flag might be able to do what I need, but I haven't been able to locate details on what "--keep" actually does.

Does "--keep" put the certificate in a well-known location that the kernel environment can access it?

**Comment by mcb30** [comment]
No, the `--keep` flag only causes iPXE to retain the downloaded file as an image (i.e. something that would be display using the [`imgstat`](https://ipxe.org/cmd/imgstat) command).  It does not automatically convey this information in any way to the kernel.

However, there is a generic mechanism within the [`initrd`](https://ipxe.org/cmd/initrd) command that allows arbitrary downloaded files to be made to appear within the initramfs as seen by the loaded kernel.  This mechanism gives you a way to make the downloaded certificate available at a path of your choice within the initramfs.  It is then entirely up to you to ensure that the initramfs chooses to do something with this file.

**Reply by gregorybruno** [reply]
Understand and I'll experiment with the `initrd` command and its flags.

Many thanks for the help!!
